File: tests/perf2/test-property-read.js
/*
 *  Basic property read performance.
 */

if (typeof window == 'undefined') window = {}
if (!window.performance) window.performance = {}
window.performance.now = window.performance.now || Date.now

if (typeof print != 'undefined') log = print
else log = console.log

function test () {
  var start
  var duration

  var obj = { xxx1: 1, xxx2: 2, xxx3: 3, xxx4: 4, foo: 123 }
  var i, ign

  start = window.performance.now()

  // begin test
  for (i = 0; i < 1e8; i++) {
    ign = obj.foo
  }
  // end test

  duration = window.performance.now() - start

  log(duration)
}

test()

File: tests/perf2/test-property-read.c
/*
 * Basic property read performance.
 */

#include "stdio.h"
#include "time.h"

typedef struct obj_t {
     int xxx1;
     int xxx2;
     int xxx3;
     int xxx4;
     int foo;
} obj_t;

int main(int argc, char **argv) {
     int msec;
     clock_t start, duration;

     obj_t obj = {
          .xxx1 = 1,
          .xxx2 = 2,
          .xxx3 = 3,
          .xxx4 = 4,
          .foo = 123
     };
     int ign;

     start = clock();

     // begin test
     for (int i = 0; i < 1e8; i++) {
          ign = obj.foo;
     }
     // end test

     duration = clock() - start;
     msec = duration * 1000 / CLOCKS_PER_SEC;

     printf("%d\n", msec);

     return 0;
}
